sources/packages/backend/apps/queue-consumers/src/processors/schedulers/esdc-integration/ecert-integration/_tests_/e-cert-utils.ts

```diff
@@ -106,3 +121,73 @@ export async function loadDisbursementSchedules(
     order: { disbursementDate: "ASC" },
   });
 }
+
+/**
+ * Creates the test data required for blocked disbursement.
+ * @param db e2e data sources.
+ * @param options options.
+ * - `offeringIntensity` offering intensity.
+ * - `isValidSIN` is student SIN valid.
+ * - `disbursementValues` disbursement values.
+ * - `msfaaState` MSFAA number approval state.
+ * @returns student and disbursement id.
+ */
+export async function createBlockedDisbursementTestData(
+  db: E2EDataSources,
+  options?: {
+    offeringIntensity?: OfferingIntensity;
+    isValidSIN?: boolean;
+    disbursementValues?: DisbursementValue[];
+    msfaaState?: MSFAAStates;
+  },
+): Promise<{
+  student: Student;
+  disbursementId: number;
+}> {
+  const offeringIntensity =
+    options?.offeringIntensity ?? OfferingIntensity.partTime;
+  const isValidSIN = options?.isValidSIN ?? false;
+  const disbursementValues = options?.disbursementValues ?? [
+    createFakeDisbursementValue(DisbursementValueType.CanadaLoan, "CSLP", 300),
+  ];
+  const msfaaState = options?.msfaaState ?? MSFAAStates.Signed;
+
+  // Student with invalid SIN to block the disbursement.
+  const student = await saveFakeStudent(db.dataSource, undefined, {
+    sinValidationInitialValue: {
+      isValidSIN,
+    },
+  });
+  // Valid MSFAA Number.
+  const msfaaNumber = await db.msfaaNumber.save(
+    createFakeMSFAANumber(
+      { student },
+      {
+        msfaaState,
+        msfaaInitialValues: {
+          offeringIntensity,
+        },
+      },
+    ),
+  );
+  // Student application.
+  const application = await saveFakeApplicationDisbursements(
+    db.dataSource,
+    {
+      student,
+      msfaaNumber,
+      disbursementValues,
+    },
+    {
+      offeringIntensity: offeringIntensity,
+      applicationStatus: ApplicationStatus.Completed,
+      firstDisbursementInitialValues: {
+        coeStatus: COEStatus.completed,
+      },
+    },
+  );
+  return {
+    student,
+    disbursementId: application.currentAssessment.disbursementSchedules[0].id,
```

Comment:
Maybe outside the PR, but because it is promoted to be shared, it would still be valid, it would be better to return the `disbursement` itself instead of its ID.